runtime/near-vm-errors/src/lib.rs

```diff
@@ -1,93 +1,130 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use near_rpc_error_macro::RpcError;
+use serde::{Deserialize, Serialize};
 use std::fmt;
 
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(
+    Debug, Clone, PartialEq, Eq, BorshDeserialize, BorshSerialize, Deserialize, Serialize, RpcError,
+)]
+#[rpc_error_variant = "FunctionCall"]
 pub enum VMError {
-    FunctionCallError(FunctionCallError),
+    FunctionExecError(FunctionExecError),
+    // TODO: serialize/deserialize?
     StorageError(Vec<u8>),
 }
 
-#[derive(Debug, Clone, PartialEq, Eq)]
-pub enum FunctionCallError {
+#[derive(
+    Debug, Clone, PartialEq, Eq, BorshDeserialize, BorshSerialize, Deserialize, Serialize, RpcError,
+)]
+#[rpc_error_variant = "FunctionExecError"]
+pub enum FunctionExecError {
     CompilationError(CompilationError),
-    LinkError(String),
+    LinkError { msg: String },
     ResolveError(MethodResolveError),
-    WasmTrap(String),
+    WasmTrap { msg: String },
     HostError(HostError),
 }
 
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(
+    Debug, Clone, PartialEq, Eq, BorshDeserialize, BorshSerialize, Deserialize, Serialize, RpcError,
+)]
+#[rpc_error_variant = "ResolveError"]
 pub enum MethodResolveError {
```

Comment:
I don't see a good reason to have a different name for it. Consistency is highly preferable. I would highly prefer if we had a strict 1:1 mapping between Rust structs and TypeScript structs without tweaks like renaming. In other words I suggest we get rid of `rpc_error_variant` param

Reply:
Why I decided to make it this way: I don't want to hard-link the name of the inner value and the variant. But I don't have a strong opinion on that..

Reply:
Let's hard link for consistency.